### PR TITLE
add design milestone requirements

### PR DESCRIPTION
**Pull request by nikhilmaddirala**

### CODE REVIEW

File: src/Paddle.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	// ================================================================================
+	// Variables
+	// ================================================================================
+	Vector2 position;
+	double halfWidth;
+	double halfHeight;
+	double speed;
+
+	// ================================================================================
```
Comment (thelmaboamah) [inline src/Paddle.java:12]: since LeftPaddle and RightPaddle have the same constructor, that code came move here too

File: src/BadObstacle.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	 * @param checkForTrigger
+	 */
+	public void triggerEvent(boolean checkForTrigger) {
+
+	}
+
+	/**
+	 * This method draws the bad obstacle.
+	 */
+	public void draw() {
```
Comment (thelmaboamah) [inline src/BadObstacle.java:31]: draw could also be moved to the parent class and just have a color param

**Review by thelmaboamah — APPROVED** [review]
Wow you had to write a lot of classes. Looks good! Left a couple comments that might help improve the design slightly, just reducing some repetition.

File: src/Paddle.java (hunk 1 above)
Comment (nikhilmaddirala) [reply src/Paddle.java:12]: Done!

File: src/BadObstacle.java (hunk 2 above)
Comment (nikhilmaddirala) [reply src/BadObstacle.java:31]: Will do in the next iteration.

**Pull request merged**